zebra/redistribute.c

```diff
@@ -184,6 +184,10 @@ void redistribute_update(const struct prefix *p, const struct prefix *src_p,
 	for (ALL_LIST_ELEMENTS(zrouter.client_list, node, nnode, client)) {
 		send_redistribute = 0;
 
+		/* Do not send unsolicited messages to synchronous clients. */
+		if (client->synchronous)
+			continue;
+
```

Comment:
Giong by commit log message. 

"Zebra is currently sending messages on interface add/delete/update,
VRF add/delete, and interface address change - regardless of whether
its clients had requested them."

Below check is meant to ensure client do not receive any route info, if not registered. 

else if (vrf_bitmap_check(client->redist[afi][ZEBRA_ROUTE_ALL],
					  re->vrf_id))

So why does it not work for you?

Reply:
You are right.  Thanks for pointing this out.

I will remove the check on line 188/189.